supervision/annotators/core.py

```diff
@@ -1496,3 +1496,138 @@ def annotate(
+        self.border_color: Color = border_color
+        self.position: Position = position
+        self.color_lookup: ColorLookup = color_lookup
+
+    def annotate(
+        self,
+        scene: np.ndarray,
+        detections: Detections,
+        custom_color_lookup: Optional[np.ndarray] = None,
+    ) -> np.ndarray:
+        """
+        Annotates the given scene with percentage bars based on the provided
+        detections.
+
+        Args:
+            scene (np.ndarray): The image where percentage bars will be drawn.
+            detections (Detections): Object detections to annotate.
+            custom_color_lookup (Optional[np.ndarray]): Custom color lookup array.
+                Allows to override the default color mapping strategy.
+
+        Returns:
+            The annotated image.
+
+        Example:
+            ```python
+            >>> import supervision as sv
+
+            >>> image = ...
+            >>> detections = sv.Detections(...)
+
+            >>> percentage_bar_annotator = sv.BoundingBoxAnnotator()
+            >>> annotated_frame = percentage_bar_annotator.annotate(
+            ...     scene=image.copy(),
+            ...     detections=detections
+            ... )
+            ```
+
+        ![percentage-bar-example](https://media.roboflow.com/
+        supervision-annotator-examples/percentage-bar-example.png)
+        """
+        for detection_idx in range(len(detections)):
+            cx, cy = detections.get_anchors_coordinates(anchor=self.position)[
+                detection_idx
+            ].astype(int)
+
+            conf = detections.confidence[detection_idx]
+            border_thickness = int(0.15 * self.height)
+
+            border_mapping = {
+                Position.TOP_LEFT: ((cx - self.width, cy - self.height), (cx, cy)),
+                Position.TOP_CENTER: (
+                    (cx - self.width // 2, cy),
+                    (cx + self.width // 2, cy - self.height),
+                ),
+                Position.TOP_RIGHT: ((cx, cy), (cx + self.width, cy - self.height)),
+                Position.CENTER_LEFT: (
+                    (cx - self.width, cy - self.height // 2),
+                    (cx, cy + self.height // 2),
+                ),
+                Position.CENTER: (
+                    (cx - self.width // 2, cy - self.height // 2),
+                    (cx + self.width // 2, cy + self.height // 2),
+                ),
+                Position.CENTER_RIGHT: (
+                    (cx, cy - self.height // 2),
+                    (cx + self.width, cy + self.height // 2),
+                ),
+                Position.BOTTOM_LEFT: ((cx - self.width, cy), (cx, cy + self.height)),
+                Position.BOTTOM_CENTER: (
+                    (cx - self.width // 2, cy),
+                    (cx + self.width // 2, cy + self.height),
+                ),
+                Position.BOTTOM_RIGHT: ((cx, cy), (cx + self.width, cy + self.height)),
```

Comment:
There is also `CENTER_OF_MASS`. It should behave as `CENTER` in this case.

Reply:
I added CENTER_OF_MASS